### PR TITLE
feat: leave when the meeting ends

### PR DESCRIPTION
**Pull request by sarahsimionescu**
complete MEE-61

### CODE REVIEW

**Comment by sarahsimionescu** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/meetingbot/meetingbot/18?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#19** <a href="https://app.graphite.dev/github/pr/meetingbot/meetingbot/19?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#18** <a href="https://app.graphite.dev/github/pr/meetingbot/meetingbot/18?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#17** <a href="https://app.graphite.dev/github/pr/meetingbot/meetingbot/17?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @sarahsimionescu and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/bots/teams/src/index.ts [hunk 1]
```diff
@@ -53,16 +53,30 @@ const file = fs.createWriteStream(__dirname + "/test.webm");
...
-  }, 1000 * 10);
 
-  // await browser.close();
+  // First wait for the leave button to appear (meaning we've joined the meeting)
+  await page.waitForSelector('button[aria-label="Leave (Ctrl+Shift+H)"]', {
+    timeout: 30000,
+  });
+  console.log("Successfully joined meeting");
+
+  // Then wait for meeting to end by watching for the "Leave" button to disappear
```
Comment (err53) [inline src/bots/teams/src/index.ts:64]: I don't think this handles the case when a meeting is ended for all participants / the bot is kicked by a user, right? Maybe we should make a note of that to handle in the future

**Review by err53 — CHANGES_REQUESTED** [review]

File: src/bots/teams/src/index.ts (hunk 1 above)
Comment (owengretzinger) [reply src/bots/teams/src/index.ts:64]: I think it should because if the bot is kicked then the leave button will disappear?

Comment (sarahsimionescu) [reply src/bots/teams/src/index.ts:64]: @owengretzinger Yes this is true!

**Comment by sarahsimionescu** [comment]
### Merge activity

* **Nov 20, 9:59 AM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/meetingbot/meetingbot/18).

**Pull request merged**